### ISSUE TITLE
[BUG] npm build is either false documented or not executed (on git dependencies)

### ISSUE THREAD

**Issue by hsjobeki** [npm/cli#5641]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

## Scripts will not be run when installing git dependencies (wrong docs? or feature disappeared?)

Docs (https://docs.npmjs.com/cli/v8/configuring-npm/package-json#git-urls-as-dependencies) claim that: 
> When installing from a git repository, the presence of certain fields in the package.json will cause npm to believe it needs to perform a build. To do so your repository will be cloned into a temporary directory, all of its deps installed, relevant scripts run, and the resulting directory packed and installed.

> This flow will occur if your git dependency uses workspaces, or if any of the following scripts are present:

> build
prepare
prepack
preinstall
install
postinstall
If your git repository includes pre-built artifacts, you will likely want to make sure that none of the above scripts are defined, or your dependency will be rebuilt for every installation.

### Expected Behavior

implement, what you wrote in the docs, or adjust the docs accordingly.

### Steps To Reproduce

I have prepared a very simple demo that can be used for testing:

https://github.com/hsjobeki/will-not-be-built
install with `npm install hsjobeki/will-not-be-built`

expected:
```js
"scripts": {
    "build": "echo \"Build step executed\" > ./test"
  },
```
so there should be a file called `test` in the folder. after installation (node_modules/will-not-build/test)

Or do i miss something?

### Environment

- npm: 8.9.0
- Node.js: v18.2.0

**Comment by lukekarrys** [comment]
Thanks for this report, I'm pretty sure that is incorrect documentation and the `build` run script should be removed from that list.

We should also audit any other places in the docs where we call out `build` in this way.

**Comment by DetachHead** [comment]
why isn't `build` run though? this prevents me from being able to use git dependencies most of the time

**Comment by DetachHead** [comment]
> all of its deps installed

it doesn't actually do this either (assuming it means `devDependencies`, otherwise the scripts are very unlikely to run successfully)

**Comment by hsjobeki** [comment]
> why isn't `build` run though? this prevents me from being able to use git dependencies most of the time

Yep… you would have to commit your build outputs into git. (Like ./dist) But i really don't like to commit generated code. 

On the other hand re-building git dependencies every time would require those to be reproducible which is also very unlikely the case.

**Comment by remcohaszing** [comment]
`prepack` isn’t executed either. This is a bug, not a documentation issue.
